### DISCUSSION TITLE
Price template: Pstryk + Enea G12

### DISCUSSION THREAD

**Discussion by q84fh** [PiotrMachowski/Home-Assistant-custom-components-TGE#14] [Price templates]
You need to prepare 4 `input_number` entities:
 - `input_number.peak_distribution_rate` - peak rate for your distribution company (in my case it is Enea with 0,45 zł/kWh as of 12.2024)
 - `input_number.offpeak_distribution_rate` - off peak rate for your distribution company (in my case it is Enea with 0.13zł/kWh as of 12.2024)
 - `input_number.market_access_fee` - this is a price for using Pstryk as your energy seller (it is 0.08zł/kWh as of 12.2024)
 - `input_number.excise`- (pl: "akcyza") - additional tax for energy (0.0045zł/kWh as of 12.2024)

```
 {# Pstryk + Enea G12 #}
 {% set s = {
     "VAT": 1.23,
     "peak_distribution_rate": states('input_number.energy_dist_peak')|float,
     "offpeak_distribution_rate": states('input_number.energy_dist_offpeak')|float,
     "market_access_fee": states('input_number.energy_pstryk_mgmt')|float,
     "excise": states('input_number.energy_excise')|float
 } %}
 
 {# Define peak periods for G12 tariff as a list of tuples (start_hour, stop_hour) #}
 {% set peaks = [(6, 13), (15, 22)] %}
 
 {# Check if the current time falls within any peak period #}
 {% set is_peak = False %}
 
 {% for peak in peaks %}
      {% if now().hour >= peak[0] and now().hour < peak[1] %}
          {% set is_peak = True %}
      {% endif %}
 {% endfor %}
 
 {# Set distribution rate based on peak or offpeak time #}
 {% set distribution_rate = s.peak_distribution_rate if is_peak else s.offpeak_distribution_rate %}
 
 {# Final price calculation #}
 {{ ((fixing1_rate + distribution_rate + s.market_access_fee) * s.VAT) + s.excise | float }}
```

**Comment by miedzeszyn** [comment]
Hi

Thanks for your code, however I am a bit confused about input.number names.
Are your names correct or should they be:

`input_number.energy_dist_peak`
`input_number.energy_dist_offpeak`
`input_number.energy_pstryk_mgmt`
`input_number.energy_excise`

?

**Reply by tomaszrondio** [reply]
you are right
